MERLin/MERLin/Routing/RouterProtocol.swift

```diff
@@ -141,51 +163,52 @@ public extension Router {
-     navigationcontroller and presented modally on the stack.
-     
-     If the shouldPush parameter is true and the "top of the stack" viewController is a
-     navigationController, then the deeplinked viewController will be pushed.
+     This  implementation of deeplink handling search for the type of the viewController starting from the top most viewController, if the
+     `from` parameter is nil, or from the `from` view controller.
+     If a responder exists for this deeplink, and a view controller is found to be of a type representing the deeplink, then the existing
+     module will be asked to update its context.
+     If the update is successful, then the deeplink is fulfilled and the updated view controller is returned, otherwise a new viewController
+     will be instanciated to represent this deeplink. This implementation will also follow new deeplink formed by the remainder of the
+     current deeplink if the `behaviour` has the smart deeplinks enabled.
      
      - parameter deeplink: The deeplink to handle
-     - parameter from: The viewController to be considered "top of the stack". If nil, the current
-     view controller will be computed in runtime.
-     - parameter shouldPush: Determine if the deeplinked viewController should be pushed or presented
-     modally in case the "top of the stack" viewController is an UINavigationController. Should push
-     also ignores updatable controllers, unless the deeplink is not updatable itself from the current deeplink
-     - parameter traverseAll: Determine if the updatable view controller should be searched in depth
-     in case of nesting of containers in the currently visible view controller. Specifically, in case of a tab bar controller
-     containing for each controller a navigation controller. With this parameter as false the search would stop to the
-     navigation controller, while with true the search will inspect the view controllers contained in the navigation controller also
+     - parameter from: The viewController to be considered "top of the stack". If nil, the current view controller will be computed
+     in runtime.
+     - parameter behaviour: The behavior of this deeplink
+     - parameter userInfo: Any additional info to be propagated to the module when the deeplink happens.
+     
+     - returns: An array of deeplinked ViewController. If smart deeplinks is enabled by the behavior this deeplink might have multiple
+     responders, and the unconsumed portion of the current deeplink string might form a new deeplink that has itself a responder to be
+     deeplinked.
      */
-    @discardableResult
-    private func handleDeeplink(_ deeplink: String, from: UIViewController? = nil, shouldPush: Bool = false, traverseAll: Bool = false, userInfo: [String: Any]?) -> [UIViewController]? {
+    internal func handle(deeplink: String, from: UIViewController?, behaviour: DeeplinkBehaviour, userInfo: [String: Any]?) -> [UIViewController] {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function should have complexity 10 or less: currently complexity equals 16](https://app.codacy.com/app/gringoireDM/MERLin/pullRequest?prid=4490332)